.github/check_agda.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+export Agda_datadir=$PWD/data
+export PATH=$PWD/bin:$PATH
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  # See https://gitlab.haskell.org/ghc/ghc/-/issues/17418
+  xattr -rc $PWD/bin
+fi
+
+echo "The nightly version of Agda is"
+agda --version
+
+echo "Checking if Agda nightly is runnable on your system..."
+for prim in ./data/lib/prim/Agda/**/*.agda
+do
+    echo "Checking "$prim
+    agda -v 2 $prim
+done
+
+if [ $? == 0 ];
+then
+  printf "$(tput bold)Agda appears to work.\nCopy the following lines to your shell profile to access the nightly build, e.g. ~/.bashrc\n"
+  echo "export Agda_datadir="$PWD"/data"
+  echo "export PATH="$PWD"/bin:\$PATH"
```

Comment:
Potential quoting issues here, especially if `PWD` has spaces which is not uncommon in Mac filenames; or apostrophes (like "`X's Code`"). You can use the bash `printf` builtin `%q` specifier to handle the quoting safely:
```diff
- echo "export Agda_datadir="$PWD"/data"
+ printf 'export Agda_datadir=%q\n' "${PWD}/data"
- echo "export PATH="$PWD"/bin:\$PATH"
+ printf 'export PATH=%q\n' "${PWD}/bin:${PATH}"
```

Reply:
`$PATH` on the second line should not be interpolated, so it becomes 

```bash
printf 'export PATH=%q:${PATH}\n' "${PWD}/bin"
```

instead.